### DISCUSSION TITLE
How to specify NAPALM credentials for each device via GUI?

### DISCUSSION THREAD

**Discussion by massivedynamics1** [netbox-community/netbox#5824] [Q&A]
Hi,

Environment:
* Django Version: 3.1.3
* Python Version: 3.6.8
* NetBox version: v2.10.4

We are testing netbox and looking for a way to enter different NAPALM credentials for each device. Is this possible via GUI or you need to use the same credentials for every device and put them into https://netbox.readthedocs.io/en/stable/configuration/optional-settings/#napalm_username ?

This seems strange if that is a way as we manage a lot of devices from different companies that obviously have different credentials for each device.

Image below:
![image](https://user-images.githubusercontent.com/53446766/108087081-5e460b80-7077-11eb-8b93-407e1749e545.png)

Thank you!

**Comment by jhammond-git** [comment]
Yes, the only place to enter NAPALM creds is in the configuration file, and you only get one.  I have heard that there's a way to specify individual creds during an API call, but as far as I know, they're only valid during that transaction.
